src/neptune_mlflow_plugin/impl/plugin.py

```diff
@@ -0,0 +1,284 @@
+        pass
+
+    def log_batch(self, run_id, metrics, params, tags):
+        neptune_run = self.get_neptune_run(run_id)
+        for metric in metrics:
+            neptune_run[f"metrics/{metric.key}"].append(metric.value, step=metric.step, timestamp=metric.timestamp)
+
+        neptune_run["params"].append({param.key: param.value for param in params})
+
+    def log_inputs(self, run_id: str, datasets: Optional[List[DatasetInput]] = None):
+        pass
+
+    def record_logged_model(self, run_id, mlflow_model):
+        pass
+
+    def log_metric(self, run_id, metric: Metric):
+        neptune_run = self.get_neptune_run(run_id)
+        neptune_run[metric.key].append(metric.value, step=metric.step, timestamp=metric.timestamp / 1e3)
+
+    def log_param(self, run_id, param: Param):
+        neptune_run = self.get_neptune_run(run_id)
+        neptune_run[param.key] = param.value
+
+    def set_tag(self, run_id, tag):
+        if tag.key:
+            neptune_tags = [tag.key, tag.value]
+        else:
+            neptune_tags = tag.value
+
+        neptune_run = self.get_neptune_run(run_id)
+        neptune_run["sys/tags"].add(neptune_tags)
+
+
+class NeptuneArtifactRepo(ArtifactRepository):
+    def __init__(self, *args, **kwargs):
+        super(NeptuneArtifactRepo, self).__init__(*args, **kwargs)
+        self._artifact_dir = local_file_uri_to_path(self.artifact_uri)
+        self._neptune_kwargs = parse_neptune_kwargs_from_uri(self.artifact_uri)
+        self._run_id = self._neptune_kwargs.pop("run_id")
+
+        # artifact-related run instances do not need to report those
+        self._neptune_kwargs.update(
+            {
+                "capture_stderr": False,
+                "capture_stdout": False,
+                "capture_hardware_metrics": False,
+            }
+        )
+
+        self._neptune_run: Optional[Run] = None
+
+    def _check_if_path_is_fileset(self, path: str) -> bool:
+        path_parts = Path(path).parts
+
+        structure = self.neptune_run.get_structure()
+
+        final_structure = structure
+        for part in path_parts:
+            final_structure = final_structure[part]
+        return isinstance(final_structure, FileSet)
+
+    @property
+    def neptune_run(self) -> Run:
+        if not self._neptune_run:
+            self._neptune_run = Run(custom_run_id=self._run_id, **self._neptune_kwargs)
+
+        return self._neptune_run
+
+    def log_artifact(self, local_file, artifact_path=None):
+        target_path = artifact_path if artifact_path else Path(local_file).stem
+
+        self.neptune_run[target_path].upload(local_file)
+
+        self.neptune_run.sync()
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        target_path = artifact_path if artifact_path else Path(local_dir).stem
+
+        self.neptune_run[target_path].upload_files(os.path.join(local_dir, "*"))
+
+        self.neptune_run.sync()
```

Comment:
I though that `upload_files` have `wait` argument that could be used instead of this `wait` call.